samples/client/petstore/kotlin-model-prefix-type-mappings/src/main/kotlin/org/openapitools/client/models/ApiApiResponse.kt

```diff
@@ -38,5 +48,85 @@ data class ApiApiResponse (
     @SerializedName("message")
     val message: kotlin.String? = null
 
-)
+) {
+
+
+    class CustomTypeAdapterFactory : TypeAdapterFactory {
+        override fun <T> create(gson: Gson, type: TypeToken<T>): TypeAdapter<T>? {
+            if (!ApiApiResponse::class.java.isAssignableFrom(type.rawType)) {
+              return null // this class only serializes 'ApiApiResponse' and its subtypes
+            }
+            val elementAdapter = gson.getAdapter(JsonElement::class.java)
+            val thisAdapter = gson.getDelegateAdapter(this, TypeToken.get(ApiApiResponse::class.java))
+
+            @Suppress("UNCHECKED_CAST")
+            return object : TypeAdapter<ApiApiResponse>() {
+                @Throws(IOException::class)
+                override fun write(out: JsonWriter, value: ApiApiResponse) {
+                    val obj = thisAdapter.toJsonTree(value).getAsJsonObject()
+                    elementAdapter.write(out, obj)
+                }
+
+                @Throws(IOException::class)
+                override fun read(jsonReader: JsonReader): ApiApiResponse  {
+                    val jsonElement = elementAdapter.read(jsonReader)
+                    validateJsonElement(jsonElement)
+                    return thisAdapter.fromJsonTree(jsonElement)
+                }
+            }.nullSafe() as TypeAdapter<T>
+        }
+    }
+
+    companion object {
+        var openapiFields: HashSet<String>? = null
+        var openapiRequiredFields: HashSet<String>? = null
```

Comment:
@wing328 if you initialise them here
```
var openapiFields = HashSet<String>()
var openapiRequiredFields = HashSet<String>()
```

it will simplify the rest of the code, because it will no longer need the `!!`
openapiFields!!.add -> openapiFields.add

Reply:
thanks for the feedback. will update accordingly.

Reply:
done